src/nile/core/account.py

```diff
@@ -64,6 +65,9 @@ def __init__(self, signer, network, predeployed_info=None):
             self.address = address
             self.index = index
 
+        # Internal address must always be of type int
+        assert type(self.address) == int
```

Comment:
is this considered internal? i expect `nre` users to access this attribute often

Reply:
True. This can be considered external, so the comment is misleading, but this address must be of type int anyway to follow the hex int pattern rule:

- Always **return int from Internal or Public API, except when the method explicitly declares the intention to return a hex** (like `nile.utils.hex_class_hash` helper).

The rationale behind this rule is to avoid parsing these values often in the source code, where we use int internally (it is simpler and more organized to convert to hex when required instead of everywhere).

Note that this rule is not for outputting to files or CLI, it is for returning values of functions or properties.

I will remove the comment, but I would let the assertion to avoid misuse of the pattern (and I'm in favor of more assertions in the code).